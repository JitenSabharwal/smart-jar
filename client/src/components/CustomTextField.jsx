import React from 'react'
import { TextField } from '@material-ui/core'

/**
 * Function creates atest input component
 * @param {object} props
 */
const CustomTextField = ({
  input,
  label,
  meta: { touched, error },
  ...custom
}) => {
  return (
    <TextField
      label={label}
      helperText={touched && error}
      fullWidth
      error={touched && error && error.length > 0}
      {...input}
      {...custom}
    />
  )
}
export default CustomTextField
