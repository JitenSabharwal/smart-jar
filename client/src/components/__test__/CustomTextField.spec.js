/* eslint-disable no-undef */
import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import CustomTextField from '../CustomTextField.jsx'
import Adapter from 'enzyme-adapter-react-16'
const obj = {
  touched: false,
  error: false,
}
Enzyme.configure({adapter: new Adapter()})
describe('CustomTextField', () => {
  test('render', () => {
    const wrapper = shallow(<CustomTextField meta={obj} />)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
