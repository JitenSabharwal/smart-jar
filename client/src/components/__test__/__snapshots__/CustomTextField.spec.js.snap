// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextField render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomTextField
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "error": false,
      "fullWidth": true,
      "helperText": false,
      "label": undefined,
      "required": false,
      "select": false,
      "variant": "standard",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "error": false,
        "fullWidth": true,
        "helperText": false,
        "label": undefined,
        "required": false,
        "select": false,
        "variant": "standard",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
